src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,57 @@
+import { useState, useRef } from 'react';
+
+declare global {
+  interface Window {
+    requestIdleCallback: (callback: () => void) => any;
+    cancelIdleCallback: (handle: any) => void;
+  }
+}
+
+// fallback: noop
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const cancelIdleCallback = window.cancelIdleCallback || (() => {});
+// fallback: execute rightaway
+const requestIdleCallback =
```

Comment:
Is this polyfilled by babel for Safari?

Reply:
babel doesn't polyfill that, since it's web plateform feature, not JavaScript. We could add a polyfill for that if we want to, or just use the fake polyfill I've added here just because this behaviour would be an improvement, but not really necessary.

Reply:
Oh I missed that. So if it's not there it just runs the function instead 👍